### PR TITLE
feat(`cheatcodes`): vm.getScriptWallets()

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes https://github.com/foundry-rs/foundry/issues/7213 + Closes https://github.com/foundry-rs/foundry/issues/6198

## Solution

Adds `vm.getScriptWallets()` that returns an array of addresses whose private keys are locally available i.e provided by the user. 

This cheatcode helps write scripts that require access to multiple wallets and use them conditionally.

### CODE REVIEW

**Comment by DaniPopes** [comment]
Are the referenced issues being fixed with this PR? If so please add the GH keywords

**Comment by yash-atreya** [comment]
> Are the referenced issues being fixed with this PR? If so please add the [GH keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).

Done

File: crates/cheatcodes/src/script.rs [hunk 1]
```diff
@@ -60,6 +61,20 @@ impl Cheatcode for stopBroadcastCall {
...
+impl Cheatcode for getScriptWalletsCall {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let script_wallets =
+            ccx.state.script_wallets().cloned().map(|sw| sw.signers().unwrap_or_default());
+
+        if let Some(script_wallets) = script_wallets {
+            let script_wallets: Vec<Address> = script_wallets.into_iter().collect();
+            Ok(script_wallets.abi_encode())
+        } else {
+            Ok("".abi_encode())
```
Comment (grandizzy) [inline crates/cheatcodes/src/script.rs:73]: can use `Ok(Default::default())`

**Review by grandizzy — COMMENTED** [review]
we could add following test in `forge/tests/cli/script.rs`

```rust
// Tests that the `getScriptWallets` cheatcode works correctly.
forgetest_init!(can_get_script_wallets, |prj, cmd| {
    let script = prj
        .add_source(
            "Foo",
            r#"
import "forge-std/Script.sol";

interface Vm {
    function getScriptWallets() external returns (address[] memory wallets);
}

contract WalletScript is Script {
    function run() public {
        address[] memory wallets = Vm(address(vm)).getScriptWallets();
        console.log(wallets[0]);
    }
}"#,
        )
        .unwrap();
    cmd.arg("script")
        .arg(script)
        .args([
            "--private-key",
            "0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6",
            "-v",
        ])
        .assert_success()
        .stdout_eq(str![[r#"
[COMPILING_FILES] with [SOLC_VERSION]
[SOLC_VERSION] [ELAPSED]
Compiler run successful!
Script ran successfully.
[GAS]

== Logs ==
  0xa0Ee7A142d267C1f36714E4a8F75612F20a79720

"#]]);
});
```

**Comment by yash-atreya** [comment]
forge-std companion PR: https://github.com/foundry-rs/forge-std/pull/620

**Review by grandizzy — APPROVED** [review]
LGTM!

File: crates/cheatcodes/src/script.rs (hunk 1 above)
Comment (FKNCOIN) [reply crates/cheatcodes/src/script.rs]: Ok (default)

**Pull request merged**